### DISCUSSION TITLE
desktop-lite using noVNC without password

### DISCUSSION THREAD

**Discussion by danielcregg** [community/community#51736] [Codespaces]
Is it possible to connect to my desktop in codespace via noVNC without using a password. Or by using a password but storing it somewhere and have it auto connect. Also is there a way to have simple browser open automatically when I open my codespace so that I can instantly see my desktop. I am developing javafx apps. 
Thanks

**Comment by samruddhikhandale** [comment]
Hi 👋 

Can you add the `desktop-lite` Feature to your devcontainer.json for this scenario? https://github.com/devcontainers/features/tree/main/src/desktop-lite has more details.

>  Also is there a way to have simple browser open automatically when I open my codespace so that I can instantly see my desktop

Can you use a `postStartCommand` for this purpose, the command runs every time the container is started (in this case, every time the codespace is connected)

**Reply by danielcregg** [reply]
Hello @samruddhikhandale, thanks for your reply. I have the desktop-lite feature enabled and it works great. Here is my repo: https://github.com/danielcregg/codespaces-javafx . 

What I am try to figure out here is how to avoid having to enter the password every time I connect to my desktop via novnc. I am thinking the solution would be to either disable the requirement for a password somewhere in the novnc configuration OR to add the password to some script somewhere so that it can be read automatically everytime I open a connection to my desktop. 

Also I cant find any reference to a postStartCommand that can control the layout of the IDE. Have you any ideas on this?

Thanks
Daniel

**Comment by superstar3222** [comment]
Hello, @danielcregg .
if you use the following link for using with the noVNC server, you can implement:
```
http://[ip-address]:[port]/vnc.html?host=[ip-address]&port=[port]&autoconnect=true&password=[your password]
```
